packages/pnpm/src/main.ts

```diff
@@ -1,10 +1,14 @@
+/* eslint-disable import/first */
+if (!global['startedAt']) {
+  global['startedAt'] = Date.now()
```

Comment:
Modifying global variables is dangerous behavior.

```suggestion
// Move to another file
export const startedAtSymbol = Symbol('startedAt')
global[startedAtSymbol] = Date.now()
```

Reply:
A better approach would be:

```ts
// In a.ts file
export const pnpmExecutionStartedAt = Date.now();

// In packages/pnpm/src/main.ts file
// line 1
import 'a.ts'

// Where it's needed
import { pnpmExecutionStartedAt } from 'a.ts';
```

Reply:
> Modifying global variables is dangerous behavior.

I know, but in this case, I don't see huge risks. Even if someone overrides the value, the worst thing that can happen is that the execution time won't be printed out correctly.

Reply:
Yes, it is true that the harm is not high in the current situation, but it would be better if there was the ability to avoid it. Especially if the solution is not complex.

Reply:
currently pnpm only has to js files. the one in bin/pnpm.cjs
and the other one is the bundled one. Now I am not sure about adding a 3rd file just for exporting the startedAt time.

Reply:
Well. It looks like this will have to do, and until we come up with something better, I agree to merge this PR